javascript/webdriver/touchsequence.js

```diff
@@ -0,0 +1,280 @@
+  return this;
+};
+
+
+/**
+ * Long press on an element.
+ *
+ * @param {webdriver.WebElement} elem The element to long press.
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.longPress = function(elem) {
+  // The interactions API expect the element ID to be encoded as a simple
+  // string, not the usual JSON object.
+  var id = /** @type {!webdriver.WebElement} */ (elem).getId().
+      then(function(value) {
+        return value['ELEMENT'];
+      });
+
+  var command = new webdriver.Command(webdriver.CommandName.TOUCH_LONG_PRESS).
+      setParameter('element', id);
+
+  this.schedule_('touchLongPress', command);
+  return this;
+};
+
+
+/**
+ * Touch down at given location.
+ *
+ * @param {x: number, y: number} location The location to touch down at.
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.tapAndHold = function(location) {
+  var command = new webdriver.Command(webdriver.CommandName.TOUCH_DOWN).
+      setParameter('x', location.x).
+      setParameter('y', location.y);
+
+  this.schedule_('touchDown', command);
+  return this;
+};
+
+
+/**
+ * Move a held touch to specified location.
+ *
+ * @param {x: number, y: number} location The location to move to.
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.move = function(location) {
+  var command = new webdriver.Command(webdriver.CommandName.TOUCH_MOVE).
+      setParameter('x', location.x).
+      setParameter('y', location.y);
+
+  this.schedule_('touchMove', command);
+  return this;
+};
+
+
+/**
+ * Release a held touch at specified location.
+ *
+ * @param {x: number, y: number} location The location to release at.
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.release = function(location) {
+  var command = new webdriver.Command(webdriver.CommandName.TOUCH_UP).
+      setParameter('x', location.x).
+      setParameter('y', location.y);
+
+  this.schedule_('touchUp', command);
+  return this;
+};
+
+
+/**
+ * Touch and scroll, moving by xoffset and yoffset.
+ *
+ * @param {xoffset: number, yoffset: number} offset The offset to scroll to.
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.scroll = function(offset) {
+  var command = new webdriver.Command(webdriver.CommandName.TOUCH_SCROLL).
+      setParameter('xoffset', offset.xoffset).
+      setParameter('yoffset', offset.yoffset);
+
+  this.schedule_('touchScroll', command);
+  return this;
+};
+
+
+/**
+ * Touch and scroll starting at elem, moving by xoffset and yoffset.
+ *
+ * @param {webdriver.WebElement} elem The element where scroll starts.
+ * @param {xoffset: number, yoffset: number} offset The offset to scroll to.
+ * @return {!webdriver.ActionSequence} A self reference.
+ */
+webdriver.TouchSequence.prototype.scrollFromElement = function(elem, offset) {
```

Comment:
Collapse scroll and scrollFromElement into a single function, like ActionSequence.mouseMove.
Also, the offset should be given as `{x: number, y: number}` and converted to xoffset/yoffset parameters internally (for consistency with ActionSequence.mouseMove)